FantasmoSDK/Classes/Utilities/LocationFuser.swift

```diff
@@ -0,0 +1,39 @@
+//
+//  LocationFuser.swift
+//  FantasmoSDK
+//
+//  Created by lucas kuzma on 8/3/21.
+//
+
+import Foundation
+import CoreLocation
+
+struct LocationFuser {
+
+    var locations: [CLLocation] = []
+
+    mutating func reset() {
+        locations = []
+    }
+
+    private func calculateConfidence(_ locations: [CLLocation]) -> FMResultConfidence {
+        switch locations.count {
+        case 1, 2:
+            return .low
+        case 3, 4:
+            return .medium
+        default:
+            return .high
+        }
+    }
+
+    mutating func fusedResult(location: CLLocation, zones: [FMZone]?) -> FMLocationResult {
```

Comment:
there's probably better name for this function since it's performing an action on a new location)

something like `getFusedResultForNew(location: CLLocation)`

docs here would help clarify either way

i also realize now that `zones` are a bit awkward since we can't fuse those.